src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseAsyncInAsync.cs

```diff
@@ -0,0 +1,265 @@
+
+        internal static DiagnosticDescriptor Descriptor = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                                      s_localizableTitle,
+                                                                                      s_localizableMessage,
+                                                                                      DiagnosticCategory.Reliability,
+                                                                                      RuleLevel.BuildWarning,
+                                                                                      s_localizableDescription,
+                                                                                      isPortedFxCopRule: false,
+                                                                                      isDataflowRule: false);
+
+        internal static DiagnosticDescriptor DescriptorNoAlternativeMethod = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                              s_localizableTitle,
+                                                                              s_localizableMessage,
+                                                                              DiagnosticCategory.Reliability,
+                                                                              RuleLevel.BuildWarning,
+                                                                              s_localizableDescription,
+                                                                              isPortedFxCopRule: false,
+                                                                              isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Descriptor);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterOperationBlockStartAction(context =>
+            {
+                context.RegisterOperationAction(AnalyzeInvocation, OperationKind.Invocation);
+
+                context.RegisterOperationAction(AnalyzePropertyGetter, OperationKind.PropertyReference);
+            });
+        }
+
+        internal static void AnalyzePropertyGetter(OperationAnalysisContext context)
+        {
+            if (IsInTaskReturningMethodOrDelegate(context))
+            {
+                InspectMemberAccess(context, CommonInterests.SyncBlockingProperties);
+            }
+        }
+
+        internal static void AnalyzeInvocation(OperationAnalysisContext context)
+        {
+            if (IsInTaskReturningMethodOrDelegate(context))
+            {
+                if (context.Operation is not IInvocationOperation)
+                {
+                    return;
```

Comment:
Can this ever happen though? You have registered only for Invocations, so you should never hit this. 1 quick way to verify is to throw an exception here instead of return and run the unit tests. If all of them pass, that means this will never be hit and you can get rid of this if block

Reply:
From the small sample size of analyzers' implementations I've reviewed, I've seen defensive code like this. But if we generally don't guard it, I'm fine being overruled here.